src/library/scala/runtime/BoxesRunTime.java

```diff
@@ -198,6 +198,20 @@ public static boolean equalsNumChar(java.lang.Number xn, java.lang.Character yc)
         }
     }
 
+    public static boolean arraysEquals(Object[] xs, Object[] ys) {
```

Comment:
Sorry, it seems I wasn't quick enough to review this before it was merged.

IMO, it is a mistake to add stuff in `BoxesRunTime` that is used only by the library, and not by the codegen. I spent a long time back in 2.12 to clean up `scala.runtime.*` from everything that shouldn't have been there, in particular moving some "utility methods" to the areas of the library where they were actually used. See https://github.com/scala/scala/pull/5098. IMO this method should be in the user-space area of the library, where it is used, and not in `scala.runtime.*`.

Reply:
Given this uses `BoxesRunTime.equals` that means moving that (and its dependants) out of `BoxesRunTime` too.

Reply:
No. In .scala code you can, and should, write `x == y` instead of `BoxesRunTime.equals(x, y)`.

Reply:
Ah, right. Thanks.

Reply:
I'm happy to move it. Where do you think it belongs?

Reply:
Sorry, @NthPortal, I got curious myself, and so ended up crafting https://github.com/scala/scala/pull/7032.